### PR TITLE
Added Developer Reference section

### PR DESCRIPTION
**Pull request by aancel**
For now, a section describing CMake has been added.

### CODE REVIEW

**Comment by prudhomm** [comment]
@aancel it reminds me that I never answered regarding external package. could you add the fonctionality  as an option eg (petsc, gmsh, ...) ?

**Comment by aancel** [comment]
@prudhomm What do you mean by that ? You want to be able to build dependencies as external packages ? I think it would be more relevant to use those on contrib packages first, don't you think ?
I already have a script for the dependencies (https://github.com/feelpp/feelpp/blob/develop/ports/linux/CMake/CMakeLists.txt), but I would have to recheck if it is working properly.

**Comment by prudhomm** [comment]
I didn't mean the libs in contrib but the ones we expect to be already installed.
I mean for boost, petsc, gmsh, cln, ...

**Comment by aancel** [comment]
Ok, then the file I previously mentioned should be doing the job. It has to be configured and installed before using the main CMakeLists.txt of Feel++

**Comment by prudhomm** [comment]
can it be on a per application/library basis rather than "just" a do it all script  ?
the do it all script would nice of course !

**Comment by aancel** [comment]
The do-it-all script is more convenient in most cases, as we can rely on find_packages in the main CMakeLists.txt.
If we use the ExternalProject directive, we'll have to specify the libraries "by hand" as they won't be available during the CMake process (They are only built at the make step).
Should we use a global option or a per-app option ? (I guess the second case is better)

File: SUMMARY.adoc [hunk 1]
```diff
@@ -80,4 +80,5 @@ Mechanics]
 . link:Annexes/README.adoc[Annexes]
 .. link:Annexes/calculus.adoc[Calculus]
 . link:GLOSSARY.adoc[Glossary]
+. link:DeveloperReference/README.adoc[Developer Reference]
```
Comment (prudhomm) [inline SUMMARY.adoc:83]: it should be moved up

**Pull request merged**